### PR TITLE
Unit tests for plugin

### PR DESCRIPTION
**Pull request by benjamreis**
Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

### CODE REVIEW

**Comment by stormi** [comment]
Suggested commit message to not have a long first line:
```
Rename parse-lsblk plugin to lsblk

- respect python namings convention
- better match what the plugin does
```

**Comment by olivierlambert** [comment]
I think in the end it might be more than that (also displaying SMART info for disks and so on), but IDK if it's too early to name it more generically 🤔

**Comment by stormi** [comment]
Then maybe something like `diskinfo`?

**Comment by olivierlambert** [comment]
What if we allow to do more than just reading info? Eg wipe the drive and so on?

**Comment by stormi** [comment]
Then maybe we could create a new plugin for actions that are not just reading info when we need it. Or choose a more generic name like `diskutil`, but it's possible that at some point we prefer to split a plugin into several smaller ones each with a well defined role. It's not easy to decide because we don't know yet precisely the scope.

**Comment by stormi** [comment]
If we change the name, then XO will need to be able to talk to both the old plugin and the new plugin. Not impossible to overcome, but if we already know that we will gather information on disks using other tools than lsblk soon, a more generic name can make sense.

**Comment by benjamreis** [comment]
For now the plugin is not released, so yeah we should decide before.

**Comment by benjamreis** [comment]
Still some work to do (i'll create issues) but it's a good 1st step and IMHO can be merged as is.

File: tests/test_hyperthreading.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            get_hyperthreading(None, None)
+            logging.error("get_hyperthreading should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
+            run_command.assert_called_with(['xl', 'info', 'threads_per_core'])
+            assert e.params[0] == '-1'
+            assert e.params[1] == 'Error!'
+            return
+        except Exception as e:
+            logging.error("Wrong exception raised: '{}', expected XenAPIPlugin.Failure".format(str(e)))
+        assert False
```
Comment (stormi) [inline tests/test_hyperthreading.py:46]: You should probably use pytest.raises instead: https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions

File: tests/test_lsblk.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        assert json.loads(res) == json.loads(expected)
+        run_command.assert_called_once_with(["lsblk", "-P", "-b", "-o", "NAME,KNAME,PKNAME,SIZE,TYPE,RO,MOUNTPOINT"])
+
+    def test_lsblk_error(self, run_command):
+        run_command.side_effect = [Exception('Error!')]
+
+        try:
+            list_block_devices(None, None)
+            logging.error("list_block_devices should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_lsblk.py:53]: Better use pytest.raises

File: tests/test_netdata.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import sys
+
+import mocked_xen_api_plugin
+import mocked_filelocker
+
+sys.modules['XenAPIPlugin'] = mocked_xen_api_plugin
+sys.modules['xcpngutils.filelocker'] = mocked_filelocker
+sys.path.append(str(pathlib.Path(__file__).parent.resolve()) + '/../SOURCES/etc/xapi.d/plugins')
+
+from netdata import is_netdata_installed, install_netdata, get_netdata_api_key
```
Comment (stormi) [inline tests/test_netdata.py:17]:
```suggestion
from netdata import is_netdata_installed, install_netdata, get_netdata_api_key

```

File: tests/test_netdata.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        run_command.assert_called_once_with(['service', 'netdata', 'restart'])
+
+    @mock.patch('netdata.run_command', autospec=True)
+    def test_install_netdata_error_at_install(self, run_command, install_package):
+        install_package.side_effect = [Exception('Error!')]
+
+        try:
+            install_netdata(None, {'api_key': 'key', 'destination': 'dest'})
+            logging.error("install_netdata should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_netdata.py:54]: Better use pytest.raises

File: tests/test_netdata.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    @mock.patch('builtins.open', mock.mock_open(read_data='1'))
+    @mock.patch('netdata.run_command', autospec=True)
+    def test_install_netdata_error_at_service(self, run_command, install_package):
+        install_package.side_effect = [None]
+        run_command.side_effect = [Exception('Error!')]
+
+        try:
+            install_netdata(None, {'api_key': 'key', 'destination': 'dest'})
+            logging.error("install_netdata should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_netdata.py:73]: Idem

File: tests/test_netdata.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        assert res == ""
+
+    @mock.patch('builtins.open', autospec=True)
+    def test_get_netdata_api_key_error(self, open):
+        open.side_effect = [Exception("Error!")]
+
+        try:
+            get_netdata_api_key(None, None)
+            logging.error("install_netdata should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_netdata.py:114]: Idem

File: tests/test_raid.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        assert json.loads(res) == json.loads(expected)
+        run_command.assert_called_once_with(['mdadm', '--detail', '/dev/md127'])
+
+    def test_raid_error(self, run_command):
+        run_command.side_effect = [Exception('Error!')]
+
+        try:
+            check_raid_pool(None, None)
+            logging.error("check_raid_pool should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_raid.py:65]: pytest.raises

File: tests/test_updater.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        assert json.loads(expected) == json.loads(res)
+
+    @mock.patch('mocked_yum.YumBase.doPackageLists', autospec=True)
+    def test_check_update_error(self, doPackageLists):
+        doPackageLists.side_effect = [Exception("Error!")]
+
+        try:
+            check_update(None, None)
+            logging.error("check_update should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_updater.py:38]: You know :)

File: tests/test_updater.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+            ['yum', 'update', '--disablerepo="*"', '--enablerepo=' + ','.join(REPOS), '-y', packages]
+        )
+
+    def test_update_error(self, run_command):
+        run_command.side_effect = [Exception("Error!")]
+
+        try:
+            update(mock.MagicMock(), {})
+            logging.error("update should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_updater.py:72]: You know

File: tests/test_updater.py [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    @mock.patch('builtins.open', autospec=True)
+    def test_set_proxies_error(self, open):
+        open.side_effect = [Exception('Error!')]
+
+        proxies = ' \
+            {"repo_yum": "http://user:password@proxy.example.com:3128", "repo_yum_2": "_none_", "repo_yum_3": "_none_"}'
+        try:
+            set_proxies(None, {"proxies": proxies})
+            logging.error("set_proxies should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_updater.py:123]: Here too

File: tests/test_zfs.py [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+        assert run_command.call_count == 3
+        run_command.assert_called_with(['zfs', 'get', '-H', 'all'])
+
+    def test_zfs_error(self, run_command):
+        run_command.side_effect = [Exception('Error!')]
+
+        try:
+            list_zfs_pools(None, None)
+            logging.error("list_zfs_pools should have raised!")
+        except mocked_xen_api_plugin.Failure as e:
```
Comment (stormi) [inline tests/test_zfs.py:126]: And here

**Review by stormi — COMMENTED** [review]
I did not go as far as analysing each test one by one.

File: SOURCES/etc/xapi.d/plugins/updater.py [hunk 12]
```diff
@@ -108,7 +105,7 @@ def check_update(session, args):
     yum_instance.populateTs(keepold=0)
     yum_instance.ts.check()  # required for ordering
     yum_instance.ts.order()
-    return json.dumps(map(display_package, packages))
+    return json.dumps(list(map(display_package, packages)))
```
[the remark this answers is not in the document]
Comment (benjamreis) [reply SOURCES/etc/xapi.d/plugins/updater.py:108]: python 2&3 compatible, will add a comment

**Review by stormi — APPROVED** [review]
Assuming that all tests do pass, approved!

**Pull request merged**